test/jdk/sun/security/krb5/auto/HttpsCB.java

```diff
@@ -1,311 +1,311 @@
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=always HttpsCB true true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=never HttpsCB false true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=invalid HttpsCB false true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          HttpsCB false true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=domain:other.com HttpsCB false true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=domain:host.web.domain HttpsCB true true
+ * @run main/othervm -Djdk.net.hosts.file=TestHosts
+ *          -Djdk.https.negotiate.cbt=domain:*.web.domain HttpsCB true true
+ */
+
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import com.sun.net.httpserver.HttpPrincipal;
+import com.sun.net.httpserver.HttpsConfigurator;
+import com.sun.net.httpserver.HttpsExchange;
+import com.sun.net.httpserver.HttpsServer;
+import com.sun.security.auth.module.Krb5LoginModule;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.PasswordAuthentication;
+import java.net.Proxy;
+import java.net.Socket;
+import java.net.URL;
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Map;
+import javax.net.ssl.HttpsURLConnection;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509ExtendedTrustManager;
+import javax.security.auth.Subject;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.net.SimpleSSLContext;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSManager;
+import sun.security.jgss.GSSUtil;
+import sun.security.jgss.krb5.internal.TlsChannelBindingImpl;
+import sun.security.krb5.Config;
+import sun.security.util.TlsChannelBinding;
+
+import java.util.Base64;
+import java.util.concurrent.Callable;
+
+public class HttpsCB {
+
+    final static String REALM_WEB = "WEB.DOMAIN";
+    final static String KRB5_CONF = "web.conf";
+    final static String KRB5_TAB = "web.ktab";
+
+    final static String WEB_USER = "web";
+    final static char[] WEB_PASS = "webby".toCharArray();
+    final static String WEB_HOST = "host.web.domain";
+    final static String CONTENT = "Hello, World!";
+
+    static int webPort;
+    static URL cbtURL;
+    static URL normalURL;
+
+    public static void main(String[] args)
+            throws Exception {
+
+        boolean expected1 = Boolean.parseBoolean(args[0]);
+        boolean expected2 = Boolean.parseBoolean(args[1]);
```

Comment:
It might be better for future maintainer and readability if these two variables could have better name, and possibly a comment to explain their purpose. AFAIU it's the expected result of running with/without CBT - where `true` means that the operation should succeed and `false` that it's expected to fail with some exception...

Reply:
Maybe `expectedCbtUrlResult` and `expectedNormalUrlResult`.